src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java

```diff
@@ -33,9 +33,11 @@
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.Map;
+import java.util.Optional;
```

Comment:
Yes we are requiring Java 8 these days. 
The year is 2022 today and the latest Java LTS is version 17.

Reply:
Fair enough, though my day job supports Java 7, and I'm still being asked to debug issues involving Java 6 in open source. Developers like to jump to the latest and greatest but customers hate this.

My comments about not using lambdas and Optional without a very good reason still apply though.

Reply:
https://maven.apache.org/developers/compatibility-plan.html
Since June 2020 Java 8 is used in Maven plugins ... so we don't stop it

Now I don't use Optional ... of course it is some type of taste, one likes Optional and other likes standard if 😄